lib/GADS.pm

```diff
@@ -1739,6 +1739,16 @@ post '/file/:id?' => require_login sub {
 # Use api route to ensure errors are returned as JSON
 post '/api/file/?' => require_login sub {
 
+    if(my $rename_id=param('rename')) {
+        my $file = schema->resultset('Fileval')->find($rename_id);
+        $file->update({name=>param('filename')});
+        return encode_json({
+            id => $file->id,
+            name => $file->name,
+            is_ok => 1,
+        });
```

Comment:
The above enables anyone with a login to rename any file. You should be able to use `find_with_permission()` to control access.

Also, please update formatting :)

Reply:
Ironically, this code is defunct anyway - I moved it over to a put request and used that in #450 (so the request could be used in either case). I vote we close this PR and use #450 in its entirety(?)

Reply:
Yes thanks, sounds good, let's do all in #450. The same will apply - there will need to be checks to ensure that the renaming aspect can only be accessed accordingly.